packages/js/product-editor/src/components/prepublish-sidebar/prepublish-sidebar.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createElement } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { PublishButton } from '../header/publish-button';
+import { PrepublishSidebarProps } from './types';
+import { store as productEditorUiStore } from '../../store/product-editor-ui';
+import { TRACKS_SOURCE } from '../../constants';
+
+export function PrepublishSidebar( {
+	productId,
+	productType = 'product',
+	title = __( 'Are you ready to add this product?', 'woocommerce' ),
+	description = __(
+		'Double-check your settings before sharing this product with customers.',
+		'woocommerce'
+	),
+}: PrepublishSidebarProps ) {
+	const lastPersistedProduct = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( 'core' );
+			return getEntityRecord( 'postType', productType, productId );
+		},
+		[ productType, productId ]
+	);
+
+	const { closePrepublishSidebar } = useDispatch( productEditorUiStore );
+
+	return (
+		<div className="woocommerce-product-publish-panel">
+			<div className="woocommerce-product-publish-panel__actions">
```

Comment:
A nitpick, but to follow inline with GB implementation maybe use `__header` here?